src/library.js

```diff
@@ -4410,8 +4416,15 @@ LibraryManager.library = {
   emscripten_get_now_is_monotonic: function() {
     // return whether emscripten_get_now is guaranteed monotonic; the Date.now
     // implementation is not :(
-    return ENVIRONMENT_IS_NODE || (typeof dateNow !== 'undefined') ||
-        ((ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) && self['performance'] && self['performance']['now']);
+    return (
+#if ENVIRONMENT_MAY_BE_NODE
+      ENVIRONMENT_IS_NODE ||
+#endif
+      (typeof dateNow !== 'undefined')
```

Comment:
I think `dateNow` is a spidermonkey detail, so this one could be behind `ENVIRONMENT_IS_SHELL` (can be a followup, however you prefer)

Reply:
Check, updated.